packages/@aws-cdk/aws-apigateway/lib/gateway-response.ts

```diff
@@ -0,0 +1,186 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnGatewayResponse } from './apigateway.generated';
+import { IRestApi } from './restapi';
+
+/**
+ * Properties for a new gateway response.
+ */
+export interface GatewayResponseProps extends GatewayResponseOptions {
+  /**
+   * Rest api resource to target.
+   */
+  readonly restApi: IRestApi;
+
+  /**
+   * Name of the gateway response.
+   * @default - Auto generated
+   */
+  readonly gatewayResponseName?: string;
+}
+
+/**
+ * Options to add gateway response.
+ */
+export interface GatewayResponseOptions {
+  /**
+   * Response type to associate with gateway response.
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/supported-gateway-response-types.html
+   */
+  readonly type: ResponseType;
+
+  /**
+   * Http status code for response.
+   * @default - standart http status code for the response type.
+   */
+  readonly statusCode?: string;
+
+  /**
+   * Custom path, query and headers for response.
+   * @default - No custom paramter will be added to response.
+   */
+  readonly parameters?: { [key: string]: string };
+
+  /**
+   * Custom templates to get mapped as response.
+   * @default - Default reponse will be mapped
+   */
+  readonly templates?: { [key: string]: string };
+
+}
+
+/**
+ * This resource creates a gateway response which will be applied to response from api gateway
+ * as per the configuration based on the response type.
+ * You will probably want to use `RestApi.addGatewayResponse` for adding new gateway response.
+ *
+ * @resource AWS::ApiGateway::GatewayResponse
+ */
+export class GatewayResponse extends Resource {
```

Comment:
For every new type of resource in CloudFormation, we usually defined an interface. This usually contains the 'Return Values' of the resource types as properties ([example](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-apigateway/lib/restapi.ts#L17)).

In this case, GatewayResponse has no 'Return Values', so let's define an empty interface.

```ts
export interface IGatewayResponse {
}
```

The `GatewayResponse` will implement this interface.